src/librustc_trans/back/linker.rs

```diff
@@ -199,6 +201,46 @@ impl<'a> Linker for GnuLinker<'a> {
     fn export_symbols(&mut self, _: &Session, _: &CrateTranslation, _: &Path) {
         // noop, visibility in object files takes care of this
     }
+
+    fn try_gold_linker(&mut self) {
+        // Only use gold under specific conditions that we know work
+
+        let gold_exists = match env::var_os("PATH") {
+            Some(ref env_path) => {
+                env::split_paths(env_path).any(|p| {
+                    let mut p = PathBuf::from(p);
```

Comment:
I think in this case `p` is already a `PathBuf` so this line shouldn't be necessary.